src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1665,16 +1642,27 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
             }
         }
 
+        if (empty($config['transports'])) {
+            $container->removeDefinition('messenger.transport.symfony_serializer');
+            $container->removeDefinition('messenger.transport.amqp.factory');
+        } else {
+            $container->getDefinition('messenger.transport.symfony_serializer')
+                ->replaceArgument(1, $config['serializer']['format'])
+                ->replaceArgument(2, $config['serializer']['context']);
+            $container->setAlias('messenger.transport.serializer', $config['serializer']['id']);
+        }
+
         $senderAliases = [];
         $transportRetryReferences = [];
         foreach ($config['transports'] as $name => $transport) {
-            if (0 === strpos($transport['dsn'], 'amqp://') && !$container->hasDefinition('messenger.transport.amqp.factory')) {
-                throw new LogicException('The default AMQP transport is not available. Make sure you have installed and enabled the Serializer component. Try enabling it or running "composer require symfony/serializer-pack".');
+            if (null !== $transport['serializer']) {
+                $transport['options']['serializer'] = new Reference($transport['serializer']);
             }
+            $serializerId = null !== $transport['serializer'] ? $transport['serializer'] : 'messenger.transport.serializer';
```

Comment:
```suggestion
            $serializerId = $transport['serializer'] ?? 'messenger.transport.serializer';
```

This would be enough isn't it?